src/pages/Detail/Detail.js

```diff
@@ -1,8 +1,104 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import { useParams } from 'react-router-dom';
+import Option from './Option';
+import ItemInfo from './ItemInfo';
+import Info from './Info';
 import './Detail.scss';
 
 const Detail = () => {
-  return <p className="">Detail</p>;
+  const { id } = useParams();
+  const [detailData, setDetailDate] = useState({});
+  const [quanItem, setQuanItem] = useState(1);
+  const [optionSwtich, setOptionSwitch] = useState(false);
+  const [option, setOption] = useState('선택');
+  const [tab, setTab] = useState('1');
+
+  useEffect(() => {
+    fetch(`/data/detail${id}.json`)
+      .then(res => res.json())
+      .then(data => setDetailDate(data));
+  }, [id]);
+  const handlePlusCount = () => {
+    setQuanItem(quanItem + 1);
+  };
+
+  const handleMinusCount = () => {
+    quanItem !== 1 ? setQuanItem(quanItem - 1) : setQuanItem(1);
+  };
+
+  const handlOption = e => {
+    setOption(e.target.value);
+    setOptionSwitch(!optionSwtich);
+  };
+  const handleTab = e => {
+    setTab(e.target.parentElement.value);
+  };
+  return (
+    <div className="detail">
+      <section className="detail_top">
+        <div className="detail_top_data">
+          <img
+            src="https://jeongyookgak-commerce.s3.ap-northeast-2.amazonaws.com/jyg-custom-seoul-app/frontend/thumbnails/transparent_background/porkbelly-fresh-detail.png"
+            alt="상품 이미지"
+          />
+
+          <div className="detail_top_content">
+            <h2 className="detail_top_title">{detailData.title}</h2>
+            <p className="detail_top_gram">100g당 3,550원</p>
+            <p className="detail_top_price">
+              기준가 {detailData.price}원({detailData.gram}g)
+            </p>
+
+            <div className="detail_top_option">
+              <span>옵션</span>
+              <div>
+                <button
+                  type="button"
+                  className="option_btn"
+                  value={option}
+                  onClick={handlOption}
+                >
+                  {option}
+                </button>
+
+                {optionSwtich ? (
+                  <Option
+                    handlOption={handlOption}
+                    option={detailData.option}
+                  />
+                ) : null}
```

Comment:
false일 때의 출력값이 null이라면 논리곱 단축평가를 활용하시면 될 것 같습니다.
```suggestion
                {optionSwtich && (
                  <Option
                    handlOption={handlOption}
                    option={detailData.option}
                  />
                )}
```

Reply:
그리고 optionSwitch보다는 boolean임을 나타내기 위해 is를 붙이는 방식으로 수정해주시면 좋을 것 같네요!

Reply:
또 handlOption에 오탈자가 있는 것 같은데 올바르게 표기해주세요! 오탈자 하나하나도 추후 유지보수에 치명적일 수 있습니다.